backend/x11/backend.c

```diff
@@ -611,33 +617,9 @@ struct wlr_backend *wlr_x11_backend_create(struct wl_display *display,
 			wlr_log(WLR_ERROR, "Failed to query DRI3 DRM FD");
 			goto error_event;
 		}
-
-		char *drm_name = drmGetDeviceNameFromFd2(x11->drm_fd);
-		wlr_log(WLR_DEBUG, "Using DRM node %s", drm_name);
-		free(drm_name);
-
-		int drm_fd = fcntl(x11->drm_fd, F_DUPFD_CLOEXEC, 0);
-		if (drm_fd < 0) {
-			wlr_log(WLR_ERROR, "fcntl(F_DUPFD_CLOEXEC) failed");
-			goto error_event;
-		}
-
-		struct wlr_gbm_allocator *gbm_alloc = wlr_gbm_allocator_create(drm_fd);
-		if (gbm_alloc == NULL) {
-			wlr_log(WLR_ERROR, "Failed to create GBM allocator");
-			close(drm_fd);
-			goto error_event;
-		}
-		x11->allocator = &gbm_alloc->base;
 		pixmap_formats = &x11->dri3_formats;
 	} else if (x11->have_shm) {
 		x11->drm_fd = -1;
-		struct wlr_shm_allocator *shm_alloc = wlr_shm_allocator_create();
-		if (shm_alloc == NULL) {
-			wlr_log(WLR_ERROR, "Failed to create shared memory allocator");
-			goto error_event;
-		}
-		x11->allocator = &shm_alloc->base;
 		pixmap_formats = &x11->shm_formats;
 	} else {
 		wlr_log(WLR_ERROR,
```

Comment:
No need for this error, `wlr_renderer_autocreate` should figure out on its own this case and return an error.